### PR TITLE
1.0.0 abi decode

### PR DESCRIPTION
**Pull request by melsomino**

### CODE REVIEW

File: ton_client/client/src/abi/decode.rs [hunk 1]
```diff
@@ -1,28 +1,48 @@
...
+use ton_sdk::AbiContract;
 
-#[derive(Serialize, Deserialize, TypeInfo)]
-pub enum MessageType {
+#[derive(Serialize, Deserialize, TypeInfo, PartialEq, Debug)]
+pub enum DecodedMessageBody {
+    /// Message has no body.
+    Empty,
+    /// Message body format is not ABI compliant.
+    Unknown,
```
Comment (AlexeyVavilin) [inline ton_client/client/src/abi/decode.rs:18]: Return error instead of `Unknown` variant

**Review by AlexeyVavilin — CHANGES_REQUESTED** [review]

File: ton_client/client/src/abi/tests.rs [hunk 2]
```diff
@@ -115,3 +116,45 @@ fn encode_v2() {
...
+            "abi.decode_message",
+            ParamsOfDecodeMessage {
+                abi: Abi::Value(events_abi.clone()),
+                message: message.into(),
+            },
+        );
+        result.body
+    };
+
+    let expected = DecodedMessageBody::FunctionInput(
```
[the remark this answers is not in the document]
Comment (melsomino) [reply ton_client/client/src/abi/tests.rs:137]: I have remove enum with parameters

File: ton_client/client/src/abi/decode.rs [hunk 3]
```diff
@@ -56,31 +41,31 @@ pub struct ParamsOfDecodeMessage {
     pub message: String,
 }
 
-use DecodedMessageBody::*;
+use MessageContentType::*;
```
Comment (AlexeyVavilin) [inline ton_client/client/src/abi/decode.rs:44]: Don't use enum variants directly. Use fully qualified name `enum_type::EnumVariant`

File: ton_client/client/src/abi/decode.rs [hunk 4]
```diff
@@ -56,31 +41,31 @@ pub struct ParamsOfDecodeMessage {
...
 ) -> ApiResult<ResultOfDecodeMessage> {
     let (abi, message) = prepare_decode(&params)?;
-    let body = if let Some(body) = message.body() {
+    let (content_type, name, value) = if let Some(body) = message.body() {
         if let Ok(output) = abi.decode_output(body.clone(), message.is_internal()) {
-            let values = get_values(&output)?;
+            let value = get_values(&output)?;
             if abi.events().get(&output.function_name).is_some() {
-                Event(output.function_name, values)
+                (Event,output.function_name, value)
```
Comment (AlexeyVavilin) [inline ton_client/client/src/abi/decode.rs]: You can return struct here without temporary tuples

**Review by AlexeyVavilin — APPROVED** [review]

**Pull request merged**